### PR TITLE
76 set legend

### PR DESCRIPTION
**Pull request by pchelle**
I have started integrating the new concepts for groupings.
I have still to work on the integration into the configuration, but the mapping seems already much cleaner

### CODE REVIEW

File: R/datamapping-groupings.R [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    color = NULL,
+    fill = NULL,
+    linetype = NULL,
+    shape = NULL,
+
+    initialize = function(color = NULL,
+                              fill = NULL,
+                              linetype = NULL,
+                              shape = NULL) {
+      if (!"GroupMapping" %in% class(color)) {
```
Comment (msevestre) [inline R/datamapping-groupings.R]: you can use isOfType here (new helper method)

File: R/datamapping-groupmapping.R [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    },
+
+    getCaptions = function(data, metaData = NULL) {
+      if (is.data.frame(self$group)) {
+        captions <- getCustomCaptions(data, self$group)
+      }
+      if (is.character(self$group)) {
+        captions <- getDefaultCaptions(data, metaData, variableList = self$group)
+      }
+      return(captions)
```
Comment (msevestre) [inline R/datamapping-groupmapping.R]: What happens here if group is neither frame nor string?  This is not complete I believe ?

File: R/plotconfiguration-axis.R [hunk 3]
```diff
@@ -2,9 +2,18 @@
...
+#'  "log", "log10", "log1p", "log2", "logit", "modulus", "probability",
+#'  "probit", "pseudo_log", "reciprocal", "reverse", "sqrt" and "time"
 Scaling <- enum(c(
+  "lin",
+  "log10",
   "log",
-  "lin"
+  "asn", "atanh", "boxcox", "date", "exp", "hms", "identity",
+  "log1p", "log2", "logit", "modulus", "probability",
+  "probit", "pseudo_log", "reciprocal", "reverse", "sqrt", "time"
```
Comment (msevestre) [inline R/plotconfiguration-axis.R]: !! WOW.

**Review by msevestre — COMMENTED** [review]
@pchelle A few minor comments. Merging

File: R/datamapping-groupmapping.R (hunk 2 above)
Comment (pchelle) [reply R/datamapping-groupmapping.R]: I added a validateIsOfType to allow only the 2 previous options

File: R/plotconfiguration-axis.R (hunk 3 above)
Comment (pchelle) [reply R/plotconfiguration-axis.R]: These are all the options available in ggplot, we can select only a few of them if we want

**Pull request merged**